src/image-copy/azext_imagecopy/custom.py

```diff
@@ -21,21 +21,28 @@ def imagecopy(cmd, source_resource_group_name, source_object_name, target_locati
               target_resource_group_name, temporary_resource_group_name='image-copy-rg',
               source_type='image', cleanup=False, parallel_degree=-1, tags=None, target_name=None,
               target_subscription=None, export_as_snapshot='false', timeout=3600):
+    only_show_errors = cmd.cli_ctx.only_show_errors
     if cleanup:
         # If --cleanup is set, forbid using an existing temporary resource group name.
         # It is dangerous to clean up an existing resource group.
         cli_cmd = prepare_cli_command(['group', 'exists', '-n', temporary_resource_group_name],
-                                      output_as_json=False)
+                                      output_as_json=False,
+                                      only_show_errors=only_show_errors)
         cmd_output = run_cli_command(cli_cmd)
         if 'true' in cmd_output:
             raise CLIError('Don\'t specify an existing resource group in --temporary-resource-group-name '
                            'when --cleanup is set')
 
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    target_subscription = get_subscription_id(cmd.cli_ctx)
+    logger.debug('subscription id - %s', target_subscription)
```

Comment:
The logic here overrides the `target_subscription` parameter entered by users. Therefore, it is necessary to add a judgment whether users have entered the `target_subscription`, such as:
```py
      if not target_subscription:
            from azure.cli.core.commands.client_factory import get_subscription_id
            target_subscription = get_subscription_id(cmd.cli_ctx)
            logger.debug('subscription id - %s', target_subscription)
```